plasma_framework/contracts/src/exits/payment/controllers/PaymentPiggybackInFlightExitOnOutput.sol

```diff
@@ -0,0 +1,146 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "../PaymentExitDataModel.sol";
+import "../PaymentInFlightExitModelLib.sol";
+import "../routers/PaymentInFlightExitRouterArgs.sol";
+import "../../IOutputGuardParser.sol";
+import "../../OutputGuardParserRegistry.sol";
+import "../../utils/ExitableTimestamp.sol";
+import "../../utils/ExitId.sol";
+import "../../utils/OutputGuard.sol";
+import "../../../framework/PlasmaFramework.sol";
+import "../../../framework/interfaces/IExitProcessor.sol";
+import "../../../transactions/outputs/PaymentOutputModel.sol";
+import "../../../transactions/PaymentTransactionModel.sol";
+import "../../../utils/IsDeposit.sol";
+import "../../../utils/UtxoPosLib.sol";
+
+library PaymentPiggybackInFlightExitOnOutput {
```

Comment:
I'm not sure the split to handle Inputs & Outputs separately was justified. Both looks very similar except the end of the `run`... I'm not sure what solidity offers to abstract just this difference - maybe simple `if` isn't that bad?

Reply:
The strongest argument was the argument is different as during input there is no need to handle outputGuard with preimage. So some args would always be empty during piggyback input.

However, in #227 we decides to make piggyback more consistent in logic.  I will take a look again.

Reply:
Merged back to one.

Reply:
ah....because of [this decision](https://github.com/omisego/plasma-contracts/pull/222#discussion_r318611936), it ends up still being inconsistence. So still break it to functions as I think it is more clean. However, I put them in same controller Lib so we can be more dry on helper functions. Also the complexity of that lib is still okay.